app/src/test/java/waiter/CommunicatorTest.java

```diff
@@ -1,28 +1,33 @@
 package waiter;
 
-import waiter.ClientConnection.Connectable;
 import waiter.Communicator.Communicator;
 import waiter.Listener.mock.ListenerMock;
 import waiter.Messenger.mock.MessengerMock;
-
-import org.junit.jupiter.api.Test;
 import waiter.Reactor.mock.ReactorMock;
 
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeEach;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 import java.io.IOException;
 
 class CommunicatorTest {
 
+    String clientHasDisconnected;
+
+    @BeforeEach
+    void setUp() {
+        clientHasDisconnected = null;
```

Comment:
Can we add some context as to why this needs to be null? This will help reviewer better understand our intent.

Reply:
Absolutely! Would doing this in the form of a comment be the way to go? Or a comment on the PR?

Reply:
I like the idea of a comment.

Reply:
Ahhhh, the (kinda) rare case of using a comment and my first comment in the project I believe, exciting! Would a comment saying something like `"The echo servers reader uses null to determine when the client should be closed to allow for reconnection/another connection"` or is that too long? Should it more be something like `"Needs to be null to get out of the client I/O loop"`

Reply:
I was think about something like "This needs to be null because reading the input stream returns null when the sockets are disconnected which signals the client is disconnected".